### PR TITLE
Chores: Enhance smart-crop code comments

### PR DESCRIPTION
**Pull request by ManyTheFish**
Enhance explanation around smart crop algorithms

### CODE REVIEW

File: milli/src/search/matches/mod.rs [hunk 1]
```diff
@@ -100,6 +101,8 @@ pub struct MatchBounds {
     pub length: usize,
 }
 
+/// Structure used to analize a string, compute words that match,
+/// and format the source string returning an highlighted and croped sub-string.
```
Comment (Kerollmops) [inline milli/src/search/matches/mod.rs:105]:
```suggestion
/// and format the source string, returning a highlighted and cropped sub-string.
```

File: milli/src/search/matches/mod.rs [hunk 2]
```diff
@@ -315,6 +325,7 @@ impl<'t, A: AsRef<[u8]>> Matcher<'t, '_, A> {
             }
         }
 
+        // finally, keep the byte index of each bounds of the crop window.
```
Comment (Kerollmops) [inline milli/src/search/matches/mod.rs:328]:
```suggestion
        // finally, keep the byte index of each bound of the crop window.
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for this documentation improvement!

File: milli/src/search/matches/mod.rs [hunk 3]
```diff
@@ -246,9 +251,14 @@ impl<'t, A: AsRef<[u8]>> Matcher<'t, '_, A> {
         // matches needs to be counted in the crop len.
         let mut remaining_words = crop_size + first_match_word_position - last_match_word_position;
 
+        // create the initial state of the crop window: 2 iterators starting from the matches positions,
+        // a reverse iterator starting from the first match token position and going trew the beginning of the text,
         let mut before_tokens = tokens[..first_match_token_position].iter().rev().peekable();
+        // an iterator starting from the last match token position and going trew the end of the text.
         let mut after_tokens = tokens[last_match_token_position..].iter().peekable();
 
+        // gross the crop window peeking in both drections
```
Comment (loiclec) [inline milli/src/search/matches/mod.rs:260]: "gross" -> "grows"

File: milli/src/search/matches/mod.rs [hunk 4]
```diff
@@ -408,13 +419,17 @@ impl<'t, A: AsRef<[u8]>> Matcher<'t, '_, A> {
         } else {
             match &self.matches {
                 Some((tokens, matches)) => {
+                    // If the text have to be croped,
```
Comment (loiclec) [inline milli/src/search/matches/mod.rs:422]: "text have" -> "text has"

File: milli/src/search/matches/mod.rs [hunk 5]
```diff
@@ -246,9 +251,14 @@ impl<'t, A: AsRef<[u8]>> Matcher<'t, '_, A> {
         // matches needs to be counted in the crop len.
         let mut remaining_words = crop_size + first_match_word_position - last_match_word_position;
 
+        // create the initial state of the crop window: 2 iterators starting from the matches positions,
+        // a reverse iterator starting from the first match token position and going trew the beginning of the text,
```
Comment (loiclec) [inline milli/src/search/matches/mod.rs:255]: I am not sure what is meant by "trew" here, maybe "towards"?

File: milli/src/search/matches/mod.rs (hunk 3 above)
Comment (ManyTheFish) [reply milli/src/search/matches/mod.rs:260]:
```suggestion
        // grows the crop window peeking in both directions
```

File: milli/src/search/matches/mod.rs [hunk 6]
```diff
@@ -113,6 +116,8 @@ pub struct Matcher<'t, 'm, A> {
 impl<'t, A: AsRef<[u8]>> Matcher<'t, '_, A> {
     /// Iterates over tokens and save any of them that matches the query.
     fn compute_matches(&mut self) -> &mut Self {
+        /// some word are counted as matches only if there are close together and in the good order,
```
[the remark this answers is not in the document]
Comment (ManyTheFish) [reply milli/src/search/matches/mod.rs:119]:
```suggestion
        /// some words are counted as matches only if they are close together and in the good order,
```

File: milli/src/search/matches/mod.rs (hunk 5 above)
Comment (ManyTheFish) [reply milli/src/search/matches/mod.rs:255]:
```suggestion
        // a reverse iterator starting from the first match token position and going towards the beginning of the text,
```

File: milli/src/search/matches/mod.rs [hunk 7]
```diff
@@ -246,9 +251,14 @@ impl<'t, A: AsRef<[u8]>> Matcher<'t, '_, A> {
         // matches needs to be counted in the crop len.
         let mut remaining_words = crop_size + first_match_word_position - last_match_word_position;
 
+        // create the initial state of the crop window: 2 iterators starting from the matches positions,
+        // a reverse iterator starting from the first match token position and going trew the beginning of the text,
         let mut before_tokens = tokens[..first_match_token_position].iter().rev().peekable();
+        // an iterator starting from the last match token position and going trew the end of the text.
```
[the remark this answers is not in the document]
Comment (ManyTheFish) [reply milli/src/search/matches/mod.rs:257]:
```suggestion
        // an iterator starting from the last match token position and going towards the end of the text.
```

**Review by loiclec — APPROVED** [review]
LGTM :)

**Review by curquiza — APPROVED** [review]
bors merge

**Pull request merged**